### PR TITLE
feat: switch to use quarkus server with keycloak on kubernetes

### PR DESCRIPTION
**Pull request by dgauldie**
There is still a lot of work to do setting this project up. But I wanted to get it in soon so that we don't have too many conflicts

### CODE REVIEW

File: kubernetes/local/gateway-httpd-deployment.yaml
Comment (Nithos) [inline kubernetes/local/gateway-httpd-deployment.yaml:9]: our proxy should not be used in this project as it is OSS and meant for others to use as well

File: kubernetes/local/gateway-httpd-htdocs.yaml
Comment (Nithos) [inline kubernetes/local/gateway-httpd-htdocs.yaml:15]: change to Terarium AUth?

File: packages/client/terarium/package.json
Comment (Nithos) [inline packages/client/terarium/package.json:7]: extract into an .env file for use in both server and client

File: packages/services/server/src/main/docker/Dockerfile.jvm
Comment (Nithos) [inline packages/services/server/src/main/docker/Dockerfile.jvm:78]: should we use temurin builds instead?

File: kubernetes/local/gateway-httpd-deployment.yaml
Comment (bigglesandginger) [reply kubernetes/local/gateway-httpd-deployment.yaml:9]: I have lost all knowledge of which docker repo names mean what.  Should we use `docker.uncharted.software` instead?  I will change to that unless someone says otherwise.

File: kubernetes/local/gateway-httpd-htdocs.yaml
Comment (bigglesandginger) [reply kubernetes/local/gateway-httpd-htdocs.yaml]: done

**Review by YohannParis — COMMENTED** [review]
This is what I got for running `quarkus dev`:

```Bash
yparis@yparis-uncharted TERArium % quarkus dev     
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.

Downloading https://services.gradle.org/distributions/gradle-7.5-bin.zip
...........10%............20%...........30%............40%...........50%............60%...........70%............80%...........90%............100%

Welcome to Gradle 7.5!

Here are the highlights of this release:
 - Support for Java 18
 - Support for building with Groovy 4
 - Much more responsive continuous builds
 - Improved diagnostics for dependency resolution

For more details see https://docs.gradle.org/7.5/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
Could not open cp_settings generic class cache for settings file '/Users/yparis/dev/TERArium/settings.gradle' (/Users/yparis/.gradle/caches/7.5/scripts/f2lxffi5r1zlou0jfzvhn2ur5).
> BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 63

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s
```

File: packages/client/terarium/package.json
Comment (dgauldie) [reply packages/client/terarium/package.json]: I don't think we need a .env file for this, but I did move specifying the port to the vite config and I set strictPort to true so that vite ensures we run on 8080 and will exit if it can't run on 8080.

Comment (Nithos) [reply packages/client/terarium/package.json]: details... still feel that it may be good to have a local config that is changeable that can affect both as they need to be in sync... but for now fine... still need to settle on the project setup for all this then see what makes the most sense

Comment (YohannParis) [reply packages/client/terarium/package.json]: Yes, I agree, once this is merge and we create a **TERArium** repo that includes scripts to install and run the whole project at different level, the port numbers should be define then

File: package.json
Comment (Nithos) [reply package.json]: renamed and removed copy

**Comment by YohannParis** [comment]
So what is missing to get this PR merged?

**Comment by bigglesandginger** [comment]
> So what is missing to get this PR merged?

I think mostly, can you run this branch successfully?  It is "done" as a preliminary workflow idea.

File: packages/client/webapp/src/utils/uncloak.ts
Comment (YohannParis) [inline packages/client/webapp/src/utils/uncloak.ts:3]: We should remove this eslint rule instead of adding this `eslint-disable-next-line`

File: packages/client/webapp/src/stores/auth.ts
Comment (YohannParis) [inline packages/client/webapp/src/stores/auth.ts:1]: Why `pinia` instead of the web api LocaleStorage?

**Review by YohannParis — DISMISSED** [review]

File: packages/client/webapp/src/utils/uncloak.ts
Comment (mwdchang) [reply packages/client/webapp/src/utils/uncloak.ts:3]: I have a set of linter rules changes in my branch for the graph and scaffolding utilities (https://github.com/DARPA-ASKEM/TERArium/tree/porting-graph-scaffold) some of which have already discussed with various folks, I think let's take the linter discussion there rather than doing this exercise multiple times.

File: packages/client/webapp/src/stores/auth.ts
Comment (Nithos) [reply packages/client/webapp/src/stores/auth.ts:1]: localstorage could be used as well but figure the app will need state management so might as well use it for this as well. works well with vue and wont accidentally forget to clear the tokens

File: packages/services/server/src/main/docker/Dockerfile.jvm
Comment (Nithos) [reply packages/services/server/src/main/docker/Dockerfile.jvm:78]: @dgauldie any input on this? 
i know the above is from their demo but just wondering if we should be consistent where possible

**Review by Nithos — COMMENTED** [review]
I think this is probably at a decent stage to merge and unblock others waiting on this

File: packages/client/webapp/src/stores/auth.ts
Comment (YohannParis) [reply packages/client/webapp/src/stores/auth.ts:1]: I see, I wasn't really keen on using a JS framework specific state management dependency, because people will start to use to bypass simple event/prop communication path, and do weird thing.

But let's move on, and I will keep an eye on it.

**Review by YohannParis — APPROVED** [review]

**Pull request merged**
